models/articles.model.js

```diff
@@ -1,5 +1,39 @@
 const db = require("../db/connection");
 
+exports.fetchArticles = (topic) => {
+  if (topic) {
+    return db
+      .query(
+        `SELECT articles.*, CAST (COUNT(comment_id) AS INT) AS comment_count 
+      FROM articles 
+      JOIN comments ON articles.article_id = comments.article_id WHERE topic = $1 GROUP BY articles.article_id ORDER BY article_id ASC;`,
+        [topic]
+      )
+      .then((data) => {
+        if (!data.rows.length) {
+          return Promise.reject({ status: 404, message: "Article not found" });
+        }
+        data.rows.forEach((index) => {
+          delete index.body; //Ticket send to remove this?
+        });
+        return data.rows;
+      });
+  } else {
+    return db
+      .query(
+        `SELECT articles.*, CAST (COUNT(comment_id) AS INT) AS comment_count 
+      FROM articles 
+      JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY article_id ASC;`
+      )
+      .then((data) => {
+        data.rows.forEach((index) => {
+          delete index.body; //Ticket send to remove this?
+        });
+        return data.rows;
+      });
+  }
+};
+
```

Comment:
This isn't very dry and we are repeating code. I understand perfectly the reasoning for your conditional logic, however could this be done in a dryer way. We could just amend the query string being passed into the database if the topic exists ("WHERE topic = $1") and concatenate the string as so. This is something that could be changed but isn't a necessity right now